include/dali/core/tensor_shape.h

```diff
@@ -1473,6 +1473,48 @@ TensorListShape<output_ndim> sample_range(const TensorListShape<ndim> &in,
   return ret;
 }
 
+/**
+ * @brief Unfolds outer dimension of TensorShape in TensorShapeList
+ * unfold_outer_dim([[2, 1], [1, 3]]) -> [1, 1, 3]
+ */
+template <int ndim>
+auto unfold_outer_dim(const TensorListShape<ndim> &shapes)
+    -> TensorListShape<ndim != DynamicDimensions ? ndim - 1 : DynamicDimensions> {
+  static_assert(ndim == DynamicDimensions || ndim >= 1,
+                "Can not unfold TensorShapeList of shapes with shape dimension 0");
+  DALI_ENFORCE(shapes.sample_dim() >= 1,
+               "Can not unfold TensorShapeList of shapes with shape dimension 0");
+
+  constexpr static int sout_dim = ndim != DynamicDimensions ? ndim - 1 : DynamicDimensions;
+  const int rout_dim = shapes.sample_dim() - 1;
+
+  int nshapes = 0;
+
+  for (int i = 0; i < shapes.size(); ++i) {
+    nshapes += shapes[i][0];
+  }
+
+  TensorListShape<sout_dim> result{nshapes, rout_dim};
```

Comment:
I'd rather avoid `{}` initialization just in case we add a constructor with `std::initializer_list` (if I had my way, we'd have had it for a long time now).

Reply:
Gottcha